src/main/java/gregtech/api/capability/impl/miner/MinerLogic.java

```diff
@@ -164,22 +162,25 @@ public void performMining() {
             NonNullList<ItemStack> blockDrops = NonNullList.create();
             IBlockState blockState = metaTileEntity.getWorld().getBlockState(blocksToMine.getFirst());
 
-            // if the block is not air or cobblestone., harvest it
-            if (GTUtility.isOre(GTUtility.toItem(blockState))) {
+            // check to make sure the ore is still there,
+            while(!GTUtility.isOre(GTUtility.toItem(blockState))) {
```

Comment:
You could optionally add new `GTUtility.isOre` function which takes `IBlockState` and calls `GTUtility.toItem` and `GTUtility.isOre` inside, so call from this side would be cleaner and more readable.